### PR TITLE
AP_Mission: Align with spec better

### PR DESCRIPTION
**Pull request by WickedShell**
Location is out of spec for NAV_RETURN_TO_LAUNCH

NAV_LOITER_UNLIM couldn't roundtrip the provided loiter radius

### CODE REVIEW

File: libraries/AP_Mission/AP_Mission.cpp [hunk 1]
```diff
@@ -994,10 +993,9 @@ bool AP_Mission::mission_cmd_to_mavlink_int(const AP_Mission::Mission_Command& c
 
     case MAV_CMD_NAV_LOITER_UNLIM:                      // MAV ID: 17
         copy_location = true;
+        packet.param3 = (float)cmd.p1;
```
Comment (magicrub) [inline libraries/AP_Mission/AP_Mission.cpp]: this does not belong, please remove

Comment (WickedShell) [inline libraries/AP_Mission/AP_Mission.cpp]: Yes it does. Unlimited has a radius, and we store it, why would we not return it?

**Comment by magicrub** [comment]
I don't understand the loiter change. Why does the current 1 or -1 scheme work? It's an integer so there shouldn't be any rounding

**Comment by magicrub** [comment]
ahh yes, you're right

**Pull request merged**